server/src/routes/donatedItemRoutes.ts

```diff
@@ -0,0 +1,65 @@
+import { Router, Request, Response } from 'express';
+import prisma from '../prismaClient'; // Import Prisma client
+import { donatedItemValidator } from '../validators/donatedItemValidator'; // Import the validator
+
+const router = Router();
+
+// POST /donatedItem - Create a new DonatedItem
+router.post('/', donatedItemValidator, async (req: Request, res: Response) => {
+    try {
+        const newItem = await prisma.donatedItem.create({
+            data: req.body,
+        });
+        console.log('New donated item created:', newItem);
+        res.status(201).json(newItem);
+    } catch (error) {
+        console.error('Error creating donated item:', error);
+        res.status(500).json({ message: 'Error creating donated item' });
+    }
+});
+
+// GET /donatedItem - Fetch all donated items
+router.get('/', async (req: Request, res: Response) => {
+    try {
+        const items = await prisma.donatedItem.findMany({
+            include: {
+                statuses: true, // Include related status updates
+            },
+        });
+        res.json(items);
+    } catch (error) {
+        console.error(error);
+        res.status(500).json({ message: 'Error fetching donated items' });
+    }
+});
+
+// PUT /donatedItem/details/:id - Update non-status details of a DonatedItem
+router.put('/details/:id', donatedItemValidator, async (req: Request, res: Response) => {
+    try {
+        const updatedItem = await prisma.donatedItem.update({
+            where: { id: Number(req.params.id) },
+            data: { ...req.body, lastUpdated: new Date() },
+        });
+        console.log('Donated item updated:', updatedItem);
+        res.json(updatedItem);
+    } catch (error) {
+        console.error('Error updating donated item details:', error);
+        res.status(500).json({ message: 'Error updating donated item details' });
+    }
+});
+
+// DELETE /donatedItem/:id - Delete a DonatedItem with cascading deletion of its statuses
+router.delete('/:id', async (req: Request, res: Response) => {
+    try {
+        const deletedItem = await prisma.donatedItem.delete({
+            where: { id: Number(req.params.id) },
+        });
+        console.log('Donated item deleted:', deletedItem);
+        res.status(200).json(deletedItem);
+    } catch (error) {
```

Comment:
Add more specific error messages for different types of errors. For instance, 404 Not Found when trying to delete or update an item that doesn't exist.
My suggestion is good to use try-catch blocks to handle different types of prisma errors.